cpp/src/examples/slice_example.cpp

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <glog/logging.h>
+#include <chrono>
+
+#include <cylon/ops.hpp>
+#include <cylon/net/mpi/mpi_communicator.hpp>
+#include <cylon/ctx/cylon_context.hpp>
+#include <cylon/table.hpp>
+
+#include "example_utils.hpp"
+
+
+int main(int argc, char *argv[]) {
+
+  if ((argc < 6 && std::string(argv[1])  == "f")) {
+    LOG(ERROR) << "./slice_example f [n | o] csv_file offset length" << std::endl
+               << "./slice_example f [n | o] csv_file  offset length" << std::endl;
+    return 1;
+  }
+
+  if ((argc < 7 && std::string(argv[1]) == "m")) {
+    LOG(ERROR) << "./slice_example m [n | o] num_tuples_per_worker 0.0-1.0 offset length" << std::endl
+               << "./slice_example m [n | o] num_tuples_per_worker 0.0-1.0 offset length" << std::endl;
+    return 1;
+  }
+  LOG(INFO) << "Starting main() function";
+  auto start_start = std::chrono::steady_clock::now();
+  auto mpi_config = std::make_shared<cylon::net::MPIConfig>();
+  
+  std::shared_ptr<cylon::CylonContext> ctx;
+  if (!cylon::CylonContext::InitDistributed(mpi_config, &ctx).is_ok()) {
+    std::cerr << "ctx init failed! " << std::endl;
+    return 1;
+  }
+
+
+  std::shared_ptr<cylon::Table> in_table, joined, sliced, head_table, tail_table;
+  auto read_options = cylon::io::config::CSVReadOptions().UseThreads(false).BlockSize(1 << 30);
```

Comment:
We should probably set default values to the `BlockSize`function and not pass in it here if it is not necessary.